modules/vba_documents/spec/factories/upload_submissions.rb

```diff
@@ -58,15 +55,24 @@
       status { 'success' }
       metadata { { 'final_success_status': Time.now.utc } }
     end
+
+    trait :skip_record_status_change_callback do
+      after(:build) do |foo|
+        class << foo
+          def record_status_change
+            true
+          end
+        end
+      end
+    end
   end
```

Comment:
This trait was added to skip some unhelpful automatic behavior in the `monthly_stats_generator_spec` which overrode the status transition times that were already set in the specs, causing the transition times to change continuously and the specs to fail. It only changes the behavior if the developer chooses to intentionally use this trait when creating records in their spec.